src/ConnectionConfig.tsx

```diff
@@ -36,11 +36,19 @@ export const ConnectionConfig: FC<ConnectionConfigProps> = (props: ConnectionCon
   if (profile === undefined) {
     profile = options.database;
   }
+  const tempCredsFeatureEnabled = config.featureToggles.awsDatasourcesTempCredentials;
 
   const settings = (window as any).grafanaBootData.settings;
   const awsAllowedAuthProviders = useMemo(
-    () => settings.awsAllowedAuthProviders ?? [AwsAuthType.Default, AwsAuthType.Keys, AwsAuthType.Credentials],
-    [settings.awsAllowedAuthProviders]
+    () =>
+      settings.awsAllowedAuthProviders ?? [
+        AwsAuthType.Default,
+        AwsAuthType.Keys,
+        AwsAuthType.Credentials,
+        AwsAuthType.GrafanaAssumeRole,
+        ...(tempCredsFeatureEnabled ? [AwsAuthType.GrafanaAssumeRole] : []),
+      ],
```

Comment:
this logic doesn't seem quite right to me? I think if you do it this way then grafanaAssumeRole will show up even when the feature is disabled.